lib/_emerge/depgraph.py

```diff
@@ -8390,7 +8398,7 @@ def _complete_graph(self, required_sets=None):
         for root in self._frozen_config.roots:
             if root != self._frozen_config.target_root and (
                 "remove" in self._dynamic_config.myparams
-                or self._frozen_config.myopts.get("--root-deps") is not None
+                or self._frozen_config.myopts.get("--root-deps") != "rdeps"
```

Comment:
Looking back at f241cdb50b0f8a448216a1e5155dba8d0abddf56, I'm not sure what I fixed. Maybe we could try removing --root-deps from this expression entirely (basically revert f241cdb50b0f8a448216a1e5155dba8d0abddf56).